company_default_partner_pricelist/models/res_company.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 ForgeFlow, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    default_property_product_pricelist = fields.Many2one(
+        "product.pricelist",
+        string="Default Account Pricelist",
+        help="Default pricelist for this company for new partners.",
+    )
+
+    def _update_partner_pricelist_generic_property(self):
+        self.ensure_one()
+        default_property_pp = self.default_property_product_pricelist
+        ir_property = self.env["ir.property"]
```

Comment:
```suggestion
        IrProperty = self.env["ir.property"]
```

Reply:
and maybe set it with `sudo()` here too since you use sudo for all the following calls